### PR TITLE
javascript-javascript1-week2

### PR DESCRIPTION
**Pull request by FurkannOzbek**
- Added txt file for codePenIo links
- Added Flight booking script 
- Added Event Application script
- Added Student Manager script
- Added Wheather wear script 
- Added Candy helper script

### CODE REVIEW

File: javascript/javascript1/week2/candy-helper.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Variables for candy prices, amount to spend, summary of candies
+
+const candyPriceSweet = 0.5;
+const candyPriceChocolate = 0.7;
+const candyPriceToffee = 1.1;
+const candyPriceChewing = 0.03;
+let boughtCandyPrices = [];
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/candy-helper.js]: If it's not very important to keep the exact price per candy spent, you could write the solution without this array and calculate sum as `sum += priceCandy`.

File: javascript/javascript1/week2/candy-helper.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+let sum = 0;
+const amountToSpend = Math.random() * 100;
+// This function calculates price of the candy & push our candy price to array
+function addCandy(candyType, weight) {
+  if (candyType === "sweet") {
+    priceCandy = weight * candyPriceSweet;
+    boughtCandyPrices.push(priceCandy);
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/candy-helper.js]:
```suggestion
    sum += priceCandy;
```
same for the other instances below

File: javascript/javascript1/week2/candy-helper.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    priceCandy = weight * candyPriceChocolate;
+    boughtCandyPrices.push(priceCandy); // Here I have tried to put push method out of all if statements but if we get wrong parameter it was also pushing the wrong parameter. So I had to put 1 by 1.
+  } else if (candyType === "toffee") {
+    priceCandy = weight * candyPriceToffee;
+    boughtCandyPrices.push(priceCandy);
+  } else if (candyType === "chewing-gum") {
+    priceCandy = weight * candyPriceChewing;
+    boughtCandyPrices.push(priceCandy);
+  } else {
+    priceCandy = console.log("Wrong parameter"); // if there is something else we get different than our parameters this error message will be shown
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/candy-helper.js:20]: That's a very good practice.

File: javascript/javascript1/week2/candy-helper.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  }
+}
+addCandy("sweet", 20);
+addCandy("toffee", 40);
+// addCandy("asd", 70); // This gives error message to console
+
+// For loop for sum the prices in the array
+for (i = 0; i < boughtCandyPrices.length; i++) {
+  sum = sum + boughtCandyPrices[i];
+}
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/candy-helper.js]: This block of code could be removed, once you utilize in the `if else` statement `sum += priceCandy`. Give it a go, of course only if you are willing to skip the `console.log` on line 45.

File: javascript/javascript1/week2/event-application.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// Event application
+
+// Created a function for calculating the day for an event creation
+function getEventWeekday(dayslater) {
+  // Fetching the system date put into a variable
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/event-application.js:5]: Commenting your code is generally a very good practice. However, it's recommended to do it for blocks of code that might not be clear from the given variable and function names. For example for complex formulas. It's a tricky balance, and a very personal thing, just don't feel required to add comments for the sake of having comments.

File: javascript/javascript1/week2/event-application.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+// Event application
+
+// Created a function for calculating the day for an event creation
+function getEventWeekday(dayslater) {
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/event-application.js]: Use camelCase for variable, properties and function names `daysLater`

File: javascript/javascript1/week2/event-application.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  // Added variable for current day name
+  const currentDay = daysOfWeek[date.getDay()];
+  // Added variable for calculating which day the events will be held
+  const eventDay = daysOfWeek[(date.getDay() + dayslater) % 7];
+  //Logging out current day and event day as text
+  console.log(`Today is ${currentDay}`);
+  console.log(`Planned event will be on ${eventDay}`);
+}
+
+// Calling the function created with parameter that I spesified
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/event-application.js]: This comment could be skipped, considering you have a comment right over where you define the function and what it does.

File: javascript/javascript1/week2/flight-booking.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+// Flight booking fullname function
+
+// When a user books a flight they write their firstname and surname, but when the ticket is printed a fullname should be displayed.
+
+// Created a function which generate a full name from firstName, surname and useFor
+function getFullName(firstName, surname, useFormalName, isMale) {
+  // First statement is checking whether given parameters are strings or not and being sure they are not empty
+  if (
+    typeof firstName && //check parameter type
+    typeof surname === "string" && //checks parameter type
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/flight-booking.js]: Unfortunately, this check won't work. You would have to explicitly write `typeof firstName === "string" && typeof surname === "string"`

File: javascript/javascript1/week2/flight-booking.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    surname.trim().length !== 0 && //removing all the spaces to check if its empty string or not
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/flight-booking.js]: What if you remove the `!== 0` part? Would it still work? Would `surname.trim().length` be considered `true` by itself, when the passed string is longer than 0 without having to explicitly check with `! == 0`?

File: javascript/javascript1/week2/flight-booking.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    firstName.trim().length !== 0 //removing all the spaces to check if its empty string or not
+  ) {
+    // Second statement is checking if useFormalName parameter is true or false for giving the lord title
+    if (useFormalName) {
+      // Third statement is checking whether man or woman
+      if (!isMale) {
+        return `Lord ${firstName} ${surname}`;
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/flight-booking.js]: You repeat `${firstName} ${surname}` three times. There is a coding principle called [DRY (Don't Repeat Yourself) ](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)which suggests creating variables and functions whenever you are to repeat a block of code more than once.

```
const fullName = `${firstName} ${surname}`;
```

File: javascript/javascript1/week2/flight-booking.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  // If type of firstName and surname parameters are not string or empty strings then returning the error message
+  else {
+    return "Wrong parameter type";
+  }
+}
+// 4 variables calling the functions has been created to check all the scenarios
+const fullname1 = getFullName("Morgan", "Freeman", true, true);
+const fullname2 = getFullName("Megan", "Fox", true, false);
+const fullname3 = getFullName("Rihanna", true, false);
+const fullname4 = getFullName("Clint", 4, true, false);
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/flight-booking.js:37]: Thank you for doing this. It helps to see that you've prepared the code for any possible scenario. Well done.

File: javascript/javascript1/week2/student-manager.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+// The statements for checking if there is more than 6 students and if the name we want to add is exist and if queen try to be in the class etc.
+// If nothing prevents it then we push our student name to the array
+function addStudentToClass(studentName) {
+  if (studentName.trim().length === 0) {
+    console.log("Please enter the name");
+  } else if (studentName === "Queen") {
+    class07Students.push("Queen");
+  } else if (class07Students.length > 5) {
+    console.log("Cannot add more student in the class");
+  } else if (class07Students.includes(studentName)) {
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/student-manager.js:12]: Beautiful!

File: javascript/javascript1/week2/student-manager.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  } else if (class07Students.length <= 5) {
+    class07Students.push(studentName);
+  }
+}
+// Function for calculate how many students in the class
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+
+console.log(getNumberOfStudents());
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/student-manager.js:23]: Do you need a function for that, when `class07Students` has global scope? You can directly console.log `class07Students.length` instead.

File: javascript/javascript1/week2/wheather-wear.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  if (tempature <= 0) {
+    return "You should wear jacket or winter coat, boots and thick clothes";
+  } else if (tempature <= 10) {
+    return " Still cold wear thick clothes";
+  } else if (tempature <= 20) {
+    return "It is little warmer you can wear less thick clothes";
+  } else if (tempature <= 30) {
+    return "Now getting there, It iss time for t-shirt and shorts";
+  } else if (tempature <= 40) {
+    return "Dont worry , most likely wont happen in Denmark in 20 years";
```
Comment (jason-vasilev) [inline javascript/javascript1/week2/wheather-wear.js:14]: Haha, I love this 😄

**Review by jason-vasilev — COMMENTED** [review]
You've done super well Furkan. Very good descriptive commits, PR and comments all across the tasks. I've left a few suggestions for cutting down on code. Please, have a look and let me know if you agree or disagree.

File: javascript/javascript1/week2/event-application.js (hunk 5 above)
Comment (FurkannOzbek) [reply javascript/javascript1/week2/event-application.js:5]: Yeah I have realized some unnecessary comments as well :D After all, most likely, who is reading code will have code knowledge  so no need to explain everything :D Thank you for pointing out and made me realize

File: javascript/javascript1/week2/student-manager.js (hunk 13 above)
Comment (FurkannOzbek) [reply javascript/javascript1/week2/student-manager.js:23]: In the description of homework It was saying copy the code and function for getNumberOfStudents() was in it , thats why I filled inside the function and I had to call it. But please let me know if I misunderstand so I can fix it

File: javascript/javascript1/week2/candy-helper.js (hunk 2 above)
Comment (FurkannOzbek) [reply javascript/javascript1/week2/candy-helper.js]: Thanks a lot, it saved me from a lot of unnecessary code ^^

File: javascript/javascript1/week2/student-manager.js (hunk 13 above)
Comment (jason-vasilev) [reply javascript/javascript1/week2/student-manager.js:23]: You are right. Just saw it as well.

**Pull request merged**